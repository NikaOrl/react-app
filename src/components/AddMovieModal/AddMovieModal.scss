@import "../../variables";

.add-modal {
  &__header {
    font-weight: 100;
    color: $white;
    padding: 0 60px 20px;
  }
}
