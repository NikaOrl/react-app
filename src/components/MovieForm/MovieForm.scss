@import "../../variables";

.movie-form {
  &__buttons-container {
    margin-left: auto;
    width: fit-content;
    padding: 0 55px 35px;

    button {
      margin: 5px;
    }
  }
}
