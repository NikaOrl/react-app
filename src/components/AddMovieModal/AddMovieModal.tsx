import React from "react";

import MovieForm from "../MovieForm/MovieForm";
import { IFilm } from "../FilmPreview/FilmPreview";

import "./AddMovieModal.scss";

interface AddMovieModalProps {
  onAdd: (film: IFilm) => void;
}

const AddMovieModal = (props: AddMovieModalProps) => {
  return (
    <div className="add-modal">
      <h1 className="add-modal__header">ADD MOVIE</h1>
      <MovieForm onSubmit={props.onAdd} />
    </div>
  );
};

export default AddMovieModal;
