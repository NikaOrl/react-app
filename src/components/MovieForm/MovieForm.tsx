import React, { ChangeEvent } from "react";
import MenuItem from "@material-ui/core/MenuItem";
import TextField from "@material-ui/core/TextField";

import Button from "../Button/Button";
import { IFilm, IFilmBase } from "../../models/film.model";

import "./MovieForm.scss";

interface MovieFormProps {
  film?: IFilm;
  onSubmit: (film: IFilm | IFilmBase) => void;
}

const genres = [
  {
    value: "Crime",
    label: "Crime"
  },
  {
    value: "Documentary",
    label: "Documentary"
  },
  {
    value: "Horror",
    label: "Horror"
  },
  {
    value: "Comedy",
    label: "Comedy"
  }
];

const MovieForm: React.FC<MovieFormProps> = (props: MovieFormProps) => {
  const [title, setTitle] = React.useState(props.film ? props.film.title : "");

  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setTitle(event.target.value);
  };

  const [selectedDate, setSelectedDate] = React.useState<string>(() => {
    if (!props.film) {
      const date = new Date();
      const month =
        date.getMonth() + 1 < 10
          ? `0${date.getMonth() + 1}`
          : date.getMonth() + 1;
      return `${date.getFullYear()}-${month}-${date.getDate()}`;
    }
    return props.film.release_date;
  });

  const handleDateChange = (
    event: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>
  ) => {
    setSelectedDate(event.target.value);
  };

  const [url, setUrl] = React.useState(
    props.film ? props.film.poster_path : ""
  );

  const handleUrlChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setUrl(event.target.value);
  };

  const [genre, setGenre] = React.useState(
    props.film ? props.film.genres.join(",") : ""
  );

  const handleGenreChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setGenre(event.target.value);
  };

  const [overview, setOverview] = React.useState(
    props.film ? props.film.overview : ""
  );

  const handleOverviewChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setOverview(event.target.value);
  };

  const [runtime, setRuntime] = React.useState(0);

  const handleRuntimeChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setRuntime(+event.target.value);
  };

  const handleSubmit = () => {
    props.onSubmit({
      id: props.film ? props.film.id : undefined,
      title: title,
      genres: [genre],
      poster_path: url,
      release_date: selectedDate,
      overview: overview,
      runtime: runtime
    });
  };

  const handleReset = () => {
    const date = new Date();
    const month =
      date.getMonth() + 1 < 10
        ? `0${date.getMonth() + 1}`
        : date.getMonth() + 1;
    const currentDate = `${date.getFullYear()}-${month}-${date.getDate()}`;
    setSelectedDate(currentDate);
    setUrl("");
    setGenre("");
    setOverview("");
    setRuntime(0);
  };

  return (
    <div className="movie-form">
      <TextField
        id="outlined-basic"
        label="TITLE"
        variant="outlined"
        value={title}
        onChange={handleTitleChange}
      />
      <TextField
        id="date"
        label="RELEASE DATE"
        type="date"
        value={selectedDate}
        onChange={handleDateChange}
        variant="outlined"
        InputLabelProps={{
          shrink: true
        }}
      />
      <TextField
        id="outlined-basic"
        label="MOVIE URL"
        variant="outlined"
        value={url}
        onChange={handleUrlChange}
      />
      <TextField
        id="outlined-basic"
        label="GENRE"
        variant="outlined"
        value={genre}
        onChange={handleGenreChange}
        select
      >
        {genres.map(option => (
          <MenuItem key={option.value} value={option.value}>
            {option.label}
          </MenuItem>
        ))}
      </TextField>
      <TextField
        id="outlined-basic"
        label="OVERVIEW"
        variant="outlined"
        value={overview}
        onChange={handleOverviewChange}
      />
      <TextField
        id="outlined-basic"
        label="RUNTIME"
        variant="outlined"
        value={runtime}
        onChange={handleRuntimeChange}
        type="number"
      />
      <div className="movie-form__buttons-container">
        <Button title="RESET" theme="grey" onClick={handleReset} />
        <Button title="SUBMIT" onClick={handleSubmit} />
      </div>
    </div>
  );
};

export default MovieForm;
