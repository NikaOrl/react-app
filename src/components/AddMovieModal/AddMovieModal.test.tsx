import React from "react";
import { render } from "@testing-library/react";
import AddMovieModal from "./AddMovieModal";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import MovieForm from "../MovieForm/MovieForm";
import { IFilm } from "../../models/film.model";
import thunk from "redux-thunk";

Enzyme.configure({ adapter: new Adapter() });

const testFilm1: IFilm = {
  id: 1,
  title: "Fifty Shades Freed",
  tagline: "Don't miss the climax",
  vote_average: 6.1,
  vote_count: 1195,
  release_date: "2018-02-07",
  poster_path:
    "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
  overview:
    "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
  budget: 55000000,
  revenue: 136906000,
  genres: ["Drama", "Romance"],
  runtime: 106
};

describe("AddMovieModal", () => {
  const initialState = { output: 10 };
  const mockStore = configureStore([thunk]);
  let store;

  it("should take a snapshot", () => {
    store = mockStore(initialState);
    store.dispatch = jest.fn();

    const mock = jest.fn();

    const { asFragment } = render(
      <Provider store={store}>
        <AddMovieModal onAdd={mock} />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should call", () => {
    store = mockStore(initialState);

    global.fetch = (jest.fn(() =>
      Promise.resolve({
        json: () => Promise.resolve({ result: "test result" })
      })
    ) as unknown) as (
      input: RequestInfo,
      init?: RequestInit | undefined
    ) => Promise<Response>;
    const mock = jest.fn();
    const wrapper = mount(
      <Provider store={store}>
        <AddMovieModal onAdd={mock} />
      </Provider>
    );
    const { onSubmit } = wrapper.find(MovieForm).props();
    onSubmit(testFilm1);
    expect(mock).toBeCalled();
  });
});
