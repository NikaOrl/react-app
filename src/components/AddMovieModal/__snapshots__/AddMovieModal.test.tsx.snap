// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMovieModal should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="add-modal"
  >
    <h1
      class="add-modal__header"
    >
      ADD MOVIE
    </h1>
    <div
      class="movie-form"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="title"
            id="title-label"
          >
            TITLE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="title"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  TITLE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="poster_path"
            id="poster_path-label"
          >
            POSTER URL
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="poster_path"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  POSTER URL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="release_date"
            id="release_date-label"
          >
            RELEASE DATE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="release_date"
              type="date"
              value="2021-04-26"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
              >
                <span>
                  RELEASE DATE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="genres"
            id="genres-label"
          >
            GENRE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="genres-label genres"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              data-testid="id-country"
              id="genres"
              role="button"
              tabindex="0"
            >
              <span>
                â€‹
              </span>
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  GENRE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="overview"
            id="overview-label"
          >
            OVERVIEW
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="overview"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  OVERVIEW
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="tagline"
            id="tagline-label"
          >
            TAGLINE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="tagline"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  TAGLINE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="vote-average"
            id="vote-average-label"
          >
            VOTE AVERAGE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="vote-average"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  VOTE AVERAGE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="vote-count"
            id="vote-count-label"
          >
            VOTE COUNT
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="vote-count"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  VOTE COUNT
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="budget"
            id="budget-label"
          >
            BUDGET
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="budget"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  BUDGET
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="revenue"
            id="revenue-label"
          >
            REVENUE
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="revenue"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3"
              >
                <span>
                  REVENUE
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink="true"
            for="runtime"
            id="runtime-label"
          >
            RUNTIME
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="runtime"
              type="number"
              value="0"
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
              >
                <span>
                  RUNTIME
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="movie-form__buttons-container"
        >
          <button
            class="btn grey"
            type="reset"
          >
            RESET
          </button>
          <button
            class="btn pink"
            type="submit"
          >
            SUBMIT
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
